### PR TITLE
Fixed #33953 -- Reverted "Fixed #33201 -- Made RenameModel operation a noop for models with db_table."

### PR DESCRIPTION
**Pull request by iurisilvio**

### CODE REVIEW

File: django/db/migrations/operations/models.py [hunk 1]
```diff
@@ -374,10 +374,9 @@ def database_forwards(self, app_label, schema_editor, from_state, to_state):
             old_db_table = old_model._meta.db_table
             new_db_table = new_model._meta.db_table
             # Don't alter when a table name is not changed.
-            if old_db_table == new_db_table:
-                return
-            # Move the main table
-            schema_editor.alter_db_table(new_model, old_db_table, new_db_table)
+            if old_db_table != new_db_table:
+                # Move the main table
+                schema_editor.alter_db_table(new_model, old_db_table, new_db_table)
```
Comment (felixxm) [inline django/db/migrations/operations/models.py:379]: I don't see much value in keeping this. `BaseDatabaseSchemaEditor.alter_db_table()` is already a noop when a table name is not changed. The whole point of afeafd6036616bac8263d762c1610f22241c0187 was in avoiding altering related fields, but we cannot do this.  

I'd revert afeafd6036616bac8263d762c1610f22241c0187 and added a new test.

**Review by felixxm — COMMENTED** [review]
@iurisilvio Thanks :+1:

File: django/db/migrations/operations/models.py (hunk 1 above)
Comment (iurisilvio) [reply django/db/migrations/operations/models.py:379]: I agree. Yesterday I was stuck trying to pass the noop test, but it'd be very ugly and probably breaking other corner cases.

Comment (felixxm) [reply django/db/migrations/operations/models.py:379]: @iurisilvio Do you have time to keep working on this? If not I can push edits according to the https://github.com/django/django/pull/15993#discussion_r954605502.

Comment (iurisilvio) [reply django/db/migrations/operations/models.py:379]: Yes, I'll update it today!

**Comment by felixxm** [comment]
@iurisilvio Thanks :+1: I added a release note.

**Pull request merged**